src/ui/component/viewers/videoViewer/view.jsx

```diff
@@ -14,16 +15,81 @@ const VIDEO_JS_OPTIONS = {
 };
 
 type Props = {
+  volume: number,
+  position: number,
+  muted: boolean,
+  hasFileInfo: boolean,
+  changeVolume: number => void,
+  savePosition: (string, number) => void,
+  changeMute: boolean => void,
+  setPlayingUri: (string | null) => void,
   source: string,
   contentType: string,
   hasFileInfo: boolean,
+  onEndedCB: any,
+  uri: string,
 };
 
 function VideoViewer(props: Props) {
-  const { contentType, source } = props;
+  const { contentType, source, setPlayingUri, onEndedCB, uri, savePosition, changeVolume, position, volume } = props;
+  console.log(volume);
   const videoRef = useRef();
   const [requireRedraw, setRequireRedraw] = useState(false);
 
+  useEffect(() => {
+    const currentVideo: HTMLVideoElement | null = document.querySelector('video');
+
+    function doEnded() {
+      // clear position
+      savePosition(uri, 0);
+      setPlayingUri(null);
+      onEndedCB();
+    }
+    function doPause(e: Event) {
+      // store position e.target.currentTime
+      if (e.target.currentTime && e.target.duration && e.target.duration - e.target.currentTime > TEN_SECONDS) {
+        savePosition(uri, e.target.currentTime);
+        console.log('store position', e.target.currentTime);
+      }
+    }
+    function doSeek() {
+      savePosition(uri, 0);
+    }
+
+    function doVolume(e: Event) {
+      // store volume e.target.volume
+      changeVolume(e.target.volume);
+      console.log('store volume', e.target.volume);
+      console.log(e);
+    }
+
+    currentVideo.currentTime = position;
+    if (currentVideo) {
+      currentVideo.volume = volume;
+    }
+
+    currentVideo.volume = volume;
+    // this didn't work
+    // currentVideo.volume = volume;
+
+    if (currentVideo) {
+      currentVideo.addEventListener('ended', doEnded);
+      currentVideo.addEventListener('pause', doPause);
+      currentVideo.addEventListener('volumechange', doVolume);
+      currentVideo.addEventListener('seeked', doSeek);
+    }
+
+    // cleanup function:
+    return () => {
+      if (currentVideo) {
+        currentVideo.removeEventListener('ended', doEnded);
+        currentVideo.removeEventListener('pause', doPause);
+        currentVideo.removeEventListener('volumechange', doVolume);
+        currentVideo.removeEventListener('seeked', doSeek);
```

Comment:
Could you just call `savePosition` here? Then every time it's taken off the page it saves. I think (?) that would do everything we need.